pact/policy-manager/policy-manager.pact

```diff
@@ -1,103 +1,64 @@
+    collection-policy:module{kip.token-policy-v2}
+    guard-policy:module{kip.token-policy-v2}
   )
 
-  (defschema ledger-guard-schema
-    guard:guard
+  (defschema ledger ;; rename to ledger-info
+    ledger:module{kip.poly-fungible-v3}
+    ledger-guard:guard
+    concrete-policies:object{concrete-policy-manager}
   )
-  
-  (deftable ledger-guard-table:{ledger-guard-schema})
 
-  (defun enforce-ledger:bool ()
-    (enforce-guard (at "guard" (read ledger-guard-table "")))
-  )
+  (deftable ledgers:{ledger})
+
+  ;; used in the filter
+  (defconst CONCRETE_POLICY_LIST
+    [NON_FUNGIBLE_POLICY QUOTE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY] )
 
-  (defcap CONCRETE_POLICY_ADMIN (policy-field:string)
-    ;;add admin check
+  (defcap ADMIN ()
     (enforce-guard 'marmalade-admin)
   )
 
   (defcap QUOTE_ESCROW (sale-id:string)
     true
   )
 
-  (defun init(marmalade-ledger-guard:guard)
-    ;;TODO adds 4 concrete policies to concrete-policy table
-    (insert ledger-guard-table "" {
-      "guard": marmalade-ledger-guard
-    })
-    true
-  )
-
-  (defun add-concrete-policy (policy-field:string policy:module{kip.token-policy-v2} )
-    (enforce (contains policy-field CONCRETE_POLICY_V1_LIST) "Not a concrete policy")
-    (with-capability (CONCRETE_POLICY_ADMIN policy-field)
-      (insert concrete-policy-table policy-field {
-        'policy-field: policy-field
-       ,'policy: policy
-      })
-    )
+  (defun enforce-ledger:bool ()
+    (enforce-guard (get-ledger-guard (get-ledger-info)))
   )
 
-  (defun get-concrete-policy:module{kip.token-policy-v2} (policy-field:string)
-    (with-read concrete-policy-table policy-field
-      {"policy":=policy }
-      policy
+  (defun init:bool(ledger-info:object{ledger})
+    (with-capability (ADMIN)
+      (insert ledgers "" ledger-info)
     )
+    true
   )
 
-  (defun get-policies-list:object{policies-list} (policies:object{token-policies})
-    (let* ( (concrete-p:[module{kip.token-policy-v2}] (create-concrete-policy-list policies))
-            (imm-p:[module{kip.token-policy-v2}] (at 'immutable-policies policies))
-            (adj-p:[module{kip.token-policy-v2}] (at 'adjustable-policies policies)) )
-      { 'concrete-policy: concrete-p
-      , 'immutable-policy: imm-p
-      , 'adjustable-policy: adj-p  } )
+  (defun get-ledger-info:object{ledger} ()
+    (read ledgers "")
   )
 
-  (defun merge-policies-list:[module{kip.token-policy-v2}] (policies:object{token-policies})
-    (let* ( (concrete-p:[module{kip.token-policy-v2}] (create-concrete-policy-list policies ))
-            (concrete-imm-p:[module{kip.token-policy-v2}] (+ concrete-p (at 'immutable-policies policies)))
-            (concrete-imm-adj-p:[module{kip.token-policy-v2}] (+ concrete-imm-p (at 'adjustable-policies policies))) )
-    concrete-imm-adj-p
-    )
+  (defun is-used:bool (policies:[module{kip.token-policy-v2}] policy:string)
+    (contains (get-concrete-policy policy) policies)
```

Comment:
I'm wondering something, we currently store the policies in list of module references (in the ledger) and we store those also in the policy-manager. 
If a policy gets updated, will the equality check still work? I've seen this fail in code before, after updating the reference wasn't equal to the stored reference in the table anymore.